### PR TITLE
Fix filter data with 0 value is not working issue

### LINKED ISSUE #10832: Unable to filter customers with 0 orders
### **Steps to reproduce the problem:**
```graphql
query Customers{
  customers(filter: {numberOfOrders: {lte: 0, gte: 0}},  first: 10){
    edges{
      node{
        id
        email
        orders{
          totalCount
        }
      }
    }
    totalCount
  }
}
```

### **Current result:**
Backend returns all customers instead of those with 0 orders

### **Expected result:**
Return all customers with 0 orders

### **Environment:**
master.staging core v3.8.0-a

### **Additional info/links:**
https://master.staging.saleor.cloud/dashboard/customers/?asc=true&sort=name&numberOfOrdersFrom=0&numberOfOrdersTo=0

### PR DESCRIPTION
**Pull request by AswinkumarRA**
I want to merge this change because to fix Unable to filter customers with 0 orders issue

fixes #10832 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by IKarbowiak** [comment]
@AswinkumarRA thank you very much for contributing! It looks good, could you only add some tests that check the behavior?

**Review by patrys — DISMISSED** [review]
We should add a regression test.

**Review by maarcingebala — APPROVED** [review]

**Comment by AswinkumarRA** [comment]
> @AswinkumarRA thank you very much for contributing! It looks good, could you only add some tests that check the behavior?

I have added test to check the behavior, please review it, Thanks

File: saleor/tests/fixtures.py [hunk 1]
```diff
@@ -3267,6 +3267,26 @@ def product_with_images(product_type, category, media_root, channel_USD):
...
+    ProductChannelListing.objects.create(
+        product=product,
+        channel=channel_USD,
+        is_published=True,
+        discounted_price_amount=0,
+        currency=channel_USD.currency_code,
+        visible_in_listings=True,
+        available_for_purchase_at=(datetime.datetime(1999, 1, 1, tzinfo=pytz.UTC)),
+    )
+    return product_list
```
Comment (patrys) [inline saleor/tests/fixtures.py:3287]: Why does this fixture take `product_list` and return `product_list` instead of the newly created product?

Comment (AswinkumarRA) [reply saleor/tests/fixtures.py:3287]: I just thought of reusing the product_list and adding one more product to it to test the bug fix

Comment (patrys) [reply saleor/tests/fixtures.py:3287]: The resulting fixture does not make sense and is not reusable. Instead, I suggest you write a separate test that explicitly discounts the product within the test body so it's clear what the test is checking.

Comment (AswinkumarRA) [reply saleor/tests/fixtures.py:3287]: sure will do that

Comment (AswinkumarRA) [reply saleor/tests/fixtures.py]: @patrys I have written a separate test as you suggested, please review it, thanks.

Comment (AswinkumarRA) [reply saleor/tests/fixtures.py]: @patrys did you get a chance to review the test?

Comment (patrys) [reply saleor/tests/fixtures.py]: I am not the right person to review it, requested a review from the core team.

Comment (AswinkumarRA) [reply saleor/tests/fixtures.py]: ok thanks

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**